source/core/work/src/main/kotlin/com/xayah/core/work/workers/AppsFastInitWorker.kt

```diff
@@ -24,18 +25,33 @@ internal class AppsFastInitWorker @AssistedInject constructor(
     private val appsRepo: AppsRepo,
 ) : CoroutineWorker(appContext, workerParams) {
     private val mNotificationBuilder by lazy { NotificationUtil.getProgressNotificationBuilder(appContext) }
+    private var mNotificationInfo: ForegroundInfo? = null
+
+    override suspend fun getForegroundInfo(): ForegroundInfo {
+        if (mNotificationInfo == null) {
+            mNotificationInfo = NotificationUtil.createForegroundInfo(
+                appContext,
+                mNotificationBuilder,
+                appContext.getString(R.string.initializing_app_list),
+                ""
+            )
+        }
+
+        return mNotificationInfo!!
+    }
 
     override suspend fun doWork(): Result = withContext(defaultDispatcher) {
         appsRepo.fastInitialize { cur, max, content ->
+            mNotificationInfo = NotificationUtil.createForegroundInfo(
+                appContext,
+                mNotificationBuilder,
+                appContext.getString(R.string.initializing_app_list),
+                content,
+                max,
+                cur
+            )
             setForeground(
```

Comment:
`cur, max, content ->` needs to be passed into `mNotificationInfo`

Reply:
It already passes when doWork() is triggered, please check the code carefully